src/pudl/workspace/datastore.py

```diff
@@ -1,715 +1,408 @@
-           each data source in the pudl.constants module. Use None for data
-           sources that do not have multiple years.
-       month (int): the month for which data should be downloaded. Only used
-           for EPA CEMS.
-       state (str): the state for which data should be downloaded. Only used
-           for EPA CEMS.
-       table (str): the table for which data should be downloaded. Only used
-           for EPA IPM.
-
-    Returns:
-       download_url (str): a full URL from which the requested data may be
-       obtained
-
-    """
-    assert_valid_param(source=source, year=year, month=month, state=state)
-
-    base_url = pc.base_data_urls[source]
-
-    if source == 'eia860':
-        if year < max(pc.data_years['eia860']):
-            download_url = f'{base_url}/archive/xls/eia860{year}.zip'
-        else:
-            download_url = f'{base_url}/xls/eia860{year}.zip'
-    elif source == 'eia861':
-        if year < 2012:
-            # Before 2012 they used 2 digit years. Y2K12 FTW!
-            download_url = f"{base_url}/f861{str(year)[2:]}.zip"
-        else:
-            download_url = f"{base_url}/f861{year}.zip"
-    elif source == 'eia923':
-        if year < 2008:
-            prefix = 'f906920_'
-        else:
-            prefix = 'f923_'
-        if year < max(pc.data_years['eia923']):
-            arch_path = 'archive/xls'
+from pathlib import Path
+import re
+import requests
+import sys
+import yaml
+
+
+# The Zenodo tokens recorded here should have read-only access to our archives.
+# Including them here is correct in order to allow public use of this tool, so
+# long as we stick to read-only keys.
+
+TOKEN = {
+    "sandbox": "WX1FW2JOrkcCvoGutoZaGS69W5P9A73wybsToLeo4nt6NP3moeIg7sPBw8nI",
+    "production": "N/A"
+}
+
+DOI = {
+    "sandbox": {
+        "eia860": "10.5072/zenodo.504556",
+        "eia861": "10.5072/zenodo.504558",
+        "eia923": "10.5072/zenodo.504560",
+        "epaipm": "10.5072/zenodo.504564",
+        "ferc1": "10.5072/zenodo.504562"
+    },
+    "production": {}
+}
+
+PUDL_YML = Path.home() / ".pudl.yml"
+
+
+class Datastore:
+    """Handle connections and downloading of Zenodo Source archives."""
+
+    def __init__(self, loglevel="WARNING", verbose=False, sandbox=False):
+        """
+        Datastore manages file retrieval for PUDL datasets.
+
+        It pulls files from Zenodo archives as needed, and caches them
+        locally.
+
+        Args:
+            loglevel: str, logging level
+            verbose: boolean. If true, logs printed to stdout
+            sandbox: boolean. If true, use the sandbox server instead of
+                     production
+        """
+        logger = logging.Logger(__name__)
+        logger.setLevel(loglevel)
+
+        if verbose:
+            logger.addHandler(logging.StreamHandler())
+
+        self.logger = logger
+        logger.info("Logging at loglevel %s" % loglevel)
```

Comment:
@cmgosnell @ptvirgo Do we collectively have a preference on string formatting methods? Personally I find f-strings significantly more readable than this kind of string interpolation, and we aren't trying to maintain backwards compatibility. Do you dislike the f-strings Pablo, or is this just a habit from working with code that needs to integrate with older versions of Python?